jpa/hibernate4_1/src/main/java/org/jboss/as/jpa/hibernate4/HibernatePersistenceProviderAdaptor.java

```diff
@@ -82,35 +77,7 @@ public void addProviderProperties(Map properties, PersistenceUnitMetadata pu) {
 
     @Override
     public void addProviderDependencies(PersistenceUnitMetadata pu) {
-        final Properties properties = pu.getProperties();
-        final String sharedCacheMode = properties.getProperty(SHARED_CACHE_MODE);
-
-        if ( Classification.NONE.equals(platform.defaultCacheClassification())) {
-            if (!SharedCacheMode.NONE.equals(pu.getSharedCacheMode())) {
-                JPA_LOGGER.tracef("second level cache is not supported in platform, ignoring shared cache mode");
-            }
-            pu.setSharedCacheMode(SharedCacheMode.NONE);
-        }
-
-        // check if 2lc is explicitly disabled which takes precedence over other settings
-        boolean sharedCacheDisabled = SharedCacheMode.NONE.equals(pu.getSharedCacheMode())
-                ||
-                NONE.equals(sharedCacheMode);
-
-        if (!sharedCacheDisabled &&
-                Boolean.parseBoolean(properties.getProperty(AvailableSettings.USE_SECOND_LEVEL_CACHE))
-                ||
-                (sharedCacheMode != null && (!NONE.equals(sharedCacheMode)))
-                || (!SharedCacheMode.NONE.equals(pu.getSharedCacheMode()) && (!SharedCacheMode.UNSPECIFIED.equals(pu.getSharedCacheMode())))) {
-            HibernateSecondLevelCache.addSecondLevelCacheDependencies(pu.getProperties(), pu.getScopedPersistenceUnitName());
-            JPA_LOGGER.tracef("second level cache enabled for %s", pu.getScopedPersistenceUnitName());
-        } else {
-            JPA_LOGGER.tracef("second level cache disabled for %s, pu %s property = %s, pu.getSharedCacheMode = %s",
-                    pu.getScopedPersistenceUnitName(),
-                    SHARED_CACHE_MODE,
-                    sharedCacheMode,
-                    pu.getSharedCacheMode().toString());
-        }
+        JPA_LOGGER.tracef("second level cache not integrated for Hibernate 4.x - %s", pu.getScopedPersistenceUnitName());
```

Comment:
Why was this support removed? Having this as a trace message seems wrong as well.

Reply:
We might want something more than a trace log message then. Probably a warning at least.